Replicator/tests/ReplParams.cc

```diff
@@ -48,7 +48,7 @@ void ReplParams::addCollections(const std::vector<C4ReplicationCollection>& coll
 
 AllocedDict ReplParams::setOptions(const AllocedDict& params, const AllocedDict& options) {
     AllocedDict result = params;
-    for ( Dict::iterator i(options); i; ++i ) {
+    for ( Dict::iterator i(options.asDict()); i; ++i ) {
         result = repl::Options::updateProperties(result, i.keyString(), i.value());
```

Comment:
What's the incentive of this? `i(options)` is very popular idiom in the code base.

Reply:
`i(options)` causes object slicing because the `Dict::iterator()` expects a `Dict`, but here we are passing `AllocedDict`

Reply:
The issue is `Dict::iterator i(options)` which takes an `AllocedDict` as a `Dict`, this causes object slicing because the additional state that `AllocedDict` holds is implicitly discarded.